src/meta/src/manager/catalog/fragment.rs

```diff
@@ -656,6 +676,62 @@ impl FragmentManager {
         Ok(())
     }
 
+    // When dropping sink into a table, there could be an unexpected meta reboot. At this time, the sink’s catalog might have been deleted,
+    // but the union branch that attaches the downstream table to the sink fragment may still exist.
+    // This could lead to issues. Therefore, we need to find the sink fragment’s downstream, then locate its union node and delete the dirty merge.
+    fn clean_dirty_table_sink_downstreams(
+        dirty_sink_into_table_upstream_fragment_id: HashSet<u32>,
+        table_fragments: &mut BTreeMapTransaction<'_, TableId, TableFragments>,
+    ) -> MetaResult<()> {
+        tracing::info!("cleaning dirty downstream merge nodes for table sink");
+
+        let mut dirty_downstream_table_ids = HashMap::new();
+        for (table_id, table_fragment) in table_fragments.tree_mut() {
+            for fragment in table_fragment.fragments.values_mut() {
```

Comment:
Directly checking `upstream_fragment_ids` in `fragment` can avoid visiting for most table fragments.

Reply:
And one other minor optimization: if the `table_id` is in the drop list, we can also skip modify it.